File: native/strapiclient/src/model/announcement.rs
use crate::model::GalleryImage;
use rustler::NifStruct;

#[derive(Debug, NifStruct)]
#[module = "Kotkowo.Announcement"]
pub struct Announcement {
    pub id: usize,
    pub title: String,
    pub tags: Vec<String>,
    pub image: GalleryImage,
}

File: lib/kotkowo/announcement.ex
defmodule Kotkowo.Announcement do
  @moduledoc false
  alias Kotkowo.GalleryImage

  @type t :: %__MODULE__{
          title: String.t(),
          image: GalleryImage,
          tags: [String.t()],
          id: pos_integer()
        }
  defstruct [:title, :image, :tags, :id]
end
